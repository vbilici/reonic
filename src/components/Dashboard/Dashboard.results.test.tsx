import { render } from '@testing-library/react';
import DashboardResults from './Dashboard.results';

describe('DashboardResults', () => {
  const mockResults = {
    totalEnergyConsumed: 100,
    theoreticalMaxDemand: 10,
    peakPowerDemand: 5,
    concurrencyFactor: 80,
  };

  it('renders the correct values', () => {
    const { getByText } = render(<DashboardResults {...mockResults} />);

    expect(getByText('Total Energy Consumed')).toBeInTheDocument();
    expect(getByText('100.00 KWh')).toBeInTheDocument();

    expect(getByText('Theoritical Max Demand')).toBeInTheDocument();
    expect(getByText('10 KW')).toBeInTheDocument();

    expect(getByText('Peak Power Demand')).toBeInTheDocument();
    expect(getByText('5 KW')).toBeInTheDocument();

    expect(getByText('Concurrency Factor')).toBeInTheDocument();
    expect(getByText('80 %')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<DashboardResults {...mockResults} />);
    expect(container).toMatchSnapshot();
  });
});