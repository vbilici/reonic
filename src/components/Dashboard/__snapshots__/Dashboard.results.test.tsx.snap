// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardResults matches snapshot 1`] = `
<div>
  <div
    class="tremor-Card-root relative w-full text-left ring-1 rounded-tremor-default p-6 bg-tremor-background ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand mx-auto"
  >
    <ul
      class="tremor-List-root w-full divide-y divide-tremor-border text-tremor-content dark:divide-dark-tremor-border dark:text-dark-tremor-content"
    >
      <li
        class="tremor-ListItem-root w-full flex justify-between items-center text-tremor-default py-2"
      >
        <span>
          Total Energy Consumed
        </span>
        <span>
          100.00
           KWh
        </span>
      </li>
      <li
        class="tremor-ListItem-root w-full flex justify-between items-center text-tremor-default py-2"
      >
        <span>
          Theoritical Max Demand
        </span>
        <span>
          10
           KW
        </span>
      </li>
      <li
        class="tremor-ListItem-root w-full flex justify-between items-center text-tremor-default py-2"
      >
        <span>
          Peak Power Demand
        </span>
        <span>
          5
           KW
        </span>
      </li>
      <li
        class="tremor-ListItem-root w-full flex justify-between items-center text-tremor-default py-2"
      >
        <span>
          Concurrency Factor
        </span>
        <span>
          80
           %
        </span>
      </li>
    </ul>
  </div>
</div>
`;
